pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/Dictionary.java

```diff
@@ -132,6 +133,13 @@ default Object getInternal(int dictId) {
 
   double getDoubleValue(int dictId);
 
+  /**
+   * NOTE: Overridden for STRING and BYTES dictionary only for now.
+   */
+  default BigDecimal getBigDecimalValue(int dictId) {
```

Comment:
Since we are adding it as native type, we should support it for all dictionaries